test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/MetricItemExtensionsTests.cs

```diff
@@ -0,0 +1,174 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Diagnostics.Metrics;
+using Moq;
+using OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation;
+using OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation.ExportClient;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Tests;
+using Xunit;
+using OtlpCollector = OpenTelemetry.Proto.Collector.Metrics.V1;
+
+namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests
+{
+    public class MetricItemExtensionsTests
+    {
+        [Fact]
+        public void TestDataPointsCountAndBatchSizeAreEqual_CreatesOnlyOneBatch()
+        {
+            var batchSize = 2;
+            var metrics = new List<Metric>();
+            using var meter = new Meter(Utils.GetCurrentMethodName());
+            using var provider = Sdk.CreateMeterProviderBuilder()
+                .AddMeter(meter.Name)
+                .AddInMemoryExporter(metrics)
+                .Build();
+
+            var dataPoints = new List<Measurement<int>>(2);
+            for (int i = 0; i < dataPoints.Capacity; i++)
+            {
+                var point = new Measurement<int>(i, new KeyValuePair<string, object>(i.ToString(), i));
+                dataPoints.Add(point);
+            }
+
+            meter.CreateObservableGauge("test_gauge", () => dataPoints);
+
+            provider.ForceFlush();
+
+            var metric = metrics.First();
+            var otlpMetrics = metric.ToOtlpMetric(batchSize);
+
+            Assert.Single(otlpMetrics);
+        }
+
+        [Fact]
+        public void TestLeftoverDataPoints_AreAlsoBatched()
+        {
+            var batchSize = 2;
+            var metrics = new List<Metric>();
+            using var meter = new Meter(Utils.GetCurrentMethodName());
+            using var provider = Sdk.CreateMeterProviderBuilder()
+                .AddMeter(meter.Name)
+                .AddInMemoryExporter(metrics)
+                .Build();
+
+            var dataPoints = new List<Measurement<int>>(3);
+            for (int i = 0; i < dataPoints.Capacity; i++)
+            {
+                var point = new Measurement<int>(i, new KeyValuePair<string, object>(i.ToString(), i));
+                dataPoints.Add(point);
+            }
+
+            meter.CreateObservableGauge("test_gauge", () => dataPoints);
+
+            provider.ForceFlush();
+            var metric = metrics.First();
+
+            var otlpMetrics = metric.ToOtlpMetric(batchSize);
+
+            Assert.Equal(batchSize, otlpMetrics.First().Gauge.DataPoints.Count);
+            Assert.Single(otlpMetrics.Last().Gauge.DataPoints);
+        }
+
+        [Fact]
+        public void TestMultipleMetrics_WithDataPointsSmallerOrEqualToBatchSize_GeneratesOnlyOneOtlpMessage()
```

Comment:
FYI, these two tests share very similar code. [XUnit supports `Theories`](https://andrewlock.net/creating-parameterised-tests-in-xunit-with-inlinedata-classdata-and-memberdata/) that can be used to pass parameters to the test and run the test with multiple inputs. This would allow you to write a single test with the details that vary (for example `BatchSize` or `NumDataPoints`) and make your code generic. 

Your test would compute the number of times that `SendExportRequest` is expected to be called and verify that.